actionpack/lib/action_controller/metal/data_streaming.rb

```diff
@@ -155,7 +156,17 @@ def send_file_headers!(options)
         disposition = options.fetch(:disposition, DEFAULT_SEND_FILE_DISPOSITION)
         unless disposition.nil?
           disposition  = disposition.to_s
-          disposition += %(; filename="#{options[:filename]}") if options[:filename]
+          filename = options[:filename]
+          if filename
+            disposition += %(; filename="#{filename}")
+
+            # Support RFC 6266
+            # http://tools.ietf.org/html/rfc6266#appendix-D
+            unless filename.ascii_only?
+              encoded_filename = CGI::escape(filename)
+              disposition += %(; filename*="#{encoded_filename.encoding}''#{encoded_filename}")
```

Comment:
I don't think these quotes should be here. Also, does `CGI.escape` cover all the characters we need it to?

Reply:
@jeremy Do you have any plan to apply your patch to Rails?

Reply:
Please do, @mtsmfm. Note that Active Storage adopted this as well (https://github.com/rails/rails/blob/master/activestorage/app/models/active_storage/filename/parameters.rb) so perhaps its implementation could be shared and reused.

Reply:
@jeremy sent a PR https://github.com/rails/rails/pull/33829